test/controllers/species_lists_controller_test.rb

```diff
@@ -274,20 +274,43 @@ def test_index_for_project_that_does_not_exist
     )
   end
 
-  def test_show_species_list
-    sl_id = species_lists(:first_species_list).id
+  def test_show_species_list_non_owner_logged_in
+    list = species_lists(:unknown_species_list)
+    observations = list.observations
+    assert(observations.any?,
+           "Need SpeciesList fixture that has >= 1 Observation")
+    assert_not_equal(rolf, list.user)
+
+    login(rolf.login)
+    get(:show, params: { id: list.id })
 
-    # Show same list with non-owner logged in.
-    login("mary")
-    get(:show, params: { id: sl_id })
     assert_template(:show)
     assert_template("comments/_comments_for_object")
+    assert_select(
+      "form:match('action', ?)",
+      %r{/observations/\d+/species_lists/#{list.id}/remove},
+      { count: 0 },
+      "Non owner should not get buttons to remove Observations from List"
+    )
+  end
+
+  def test_show_species_list_owner_logged_in
+    list = species_lists(:unknown_species_list)
+    observations = list.observations
+    assert(observations.any?,
+           "Need SpeciesList fixture that has >= 1 Observation")
+
+    login(list.user.login)
+    get(:show, params: { id: list.id })
 
-    # Show non-empty list with owner logged in.
-    get(:show,
-        params: { id: species_lists(:unknown_species_list).id })
     assert_template(:show)
     assert_template("comments/_comments_for_object")
+    assert_select(
+      "form:match('action', ?)",
+      %r{/observations/\d+/species_lists/#{list.id}/remove},
+      { count: 2 },
```

Comment:
Seems like it would be better to use `observations.count` rather than the magic constant `2`.

Reply:
Oops. That's what I meant.